src/main/kotlin/de/livepoll/api/service/PollItemService.kt

```diff
@@ -143,21 +150,59 @@ class PollItemService {
 
-    fun updateMultipleChoiceItem(pollItemId: Long, pollItem: MultipleChoiceItemDtoIn): MultipleChoiceItemDtoOut {
+    /**
+     * Move poll item to another position while updating the positions of other poll items in the poll.<br>
+     *
+     * This method works in-place and will adjust the poll items list.<br>
+     * Old and new position are counted from 1 onwards, NOT from 0!
+     */
+    fun movePollItem(oldPos: Int, newPos: Int, pollItems: MutableList<PollItem>) {
+        // Check if new position is existent
+        if (!(newPos >= 1 && newPos <= pollItems.size)) {
+            throw ResponseStatusException(HttpStatus.BAD_REQUEST, "Invalid poll item position")
+        }
+
+        // Sort poll items by position instead of id
+        pollItems.sortBy { it.position }
+
+        // Update elements in between
+        if (oldPos < newPos) {
+            for (i in oldPos + 1..newPos) { // For every element in [oldPos+1, newPos]
+                pollItems[i - 1].position-- // Decrease position by one
+                pollItemRepository.saveAndFlush(pollItems[i - 1])
+            }
+        } else if (oldPos > newPos) {
+            for (i in newPos until oldPos) {  // For every element in [newPos, oldPos-1]
+                pollItems[i - 1].position++ // Increase position by one
+                pollItemRepository.saveAndFlush(pollItems[i - 1])
+            }
+        } else { // oldPos == newPos
+            // do nothing
+        }
+
+        // Update position of explicitly requested poll item
+        pollItems[oldPos - 1].position = newPos
+        pollItemRepository.saveAndFlush(pollItems[oldPos - 1])
+    }
+
+    fun updateMultipleChoiceItem(
+        pollItemId: Long,
+        pollItem: MultipleChoiceItemWithPositionDtoIn
+    ): MultipleChoiceItemDtoOut {
         multipleChoiceItemRepository.findById(pollItemId)
             .orElseThrow { ResponseStatusException(HttpStatus.NOT_FOUND, "Poll item not found") }
             .run {
                 val newAnswers = pollItem.answers.toMutableList()
                 val removeAnswer = mutableListOf<MultipleChoiceItemAnswer>()
                 this.answers.forEach {
                     if (it.answerCount != 0) {
-                        if(pollItem.answers.contains(it.selectionOption)){
+                        if (pollItem.answers.contains(it.selectionOption)) {
                             newAnswers.remove(it.selectionOption)
-                        }else{
+                        } else {
                             this.answers.remove(it)
                         }
-                    }else{
-                        if(!pollItem.answers.contains(it.selectionOption)){
+                    } else {
+                        if (!pollItem.answers.contains(it.selectionOption)) {
                             removeAnswer.add(it)
                         }
                     }
```

Comment:
The adjustments so that the logic works correctly are missing here. Did you forget them or do you want to add them in a future commit?

Reply:
I purposely left them out to add them to another (future) branch.